### PR TITLE
Weaken the condition for opening the external editor to any graphical…

### PR DESCRIPTION
**Pull request by adrian-prantl**
… session.

### CODE REVIEW

**Review by JDevlieghere — APPROVED** [review]
Nice

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by kastiglione — APPROVED** [review]

File: lldb/source/Commands/CommandObjectHealthcheck.cpp [hunk 1]
```diff
@@ -53,11 +53,10 @@ bool CommandObjectHealthcheck::DoExecute(Args &args,
...
-  // an interactive graphical session and not, for example, LLDB
-  // running over ssh. Open the log file straight away in the user's
-  // configured editor or Console.app otherwise.
-  if (strcmp("lldb-rpc-server", getprogname()) == 0)
+  // When in an interactive graphical session and not, for example,
+  // running LLDB running over ssh, open the log file straight away in
+  // the user's configured editor or the default Console.app otherwise.
+  if (Host::IsInteractiveGraphicSession())
     Host::OpenFileInExternalEditor(FileSpec(temp_path), 0);
 #endif
```
Comment (JDevlieghere) [inline lldb/source/Commands/CommandObjectHealthcheck.cpp:61]: Actually, this define is not need. Both Host functions exist on non-Apple platform, and will just return false or be a NOOP respectively.

Comment (adrian-prantl) [reply lldb/source/Commands/CommandObjectHealthcheck.cpp:61]: https://github.com/apple/llvm-project/pull/4288

**Pull request merged**